src/Features/CSharp/Portable/Completion/CompletionProviders/SpeculativeTCompletionProvider.cs

```diff
@@ -47,6 +47,11 @@ public override bool IsTriggerCharacter(SourceText text, int characterPosition,
 
         private async Task<bool> ShouldShowSpeculativeTCompletionItemAsync(Document document, int position, CancellationToken cancellationToken)
         {
+            if (!document.SupportsSyntaxTree)
```

Comment:
This should only ever be pulled into a C# context... And, in that case, SupportsSyntaxTree would be true... so why is htis needed.